psi_cardinality/deps.bzl

```diff
@@ -29,13 +29,7 @@ def psi_cardinality_deps():
     if "com_tencent_rapidjson" not in native.existing_rules():
         http_archive(
             name = "com_tencent_rapidjson",
-            build_file_content = """
-cc_library(
-    name = "json",
-    hdrs = glob(["**"]),
-    visibility = ["//visibility:public"],
-)
-    """,
+            build_file = "//third_party:rapidjson.BUILD",
             sha256 = "e6fc99c7df7f29995838a764dd68df87b71db360f7727ace467b21b82c85efda",
             strip_prefix = "rapidjson-8f4c021fa2f1e001d2376095928fc0532adf2ae6/include",
             url = "https://github.com/Tencent/rapidjson/archive/8f4c021fa2f1e001d2376095928fc0532adf2ae6.zip",
```

Comment:
isn't a bit weird that the download path for a third party library is in the psi_cardinality folder?

Reply:
Hmm yeah, I was again following convention to put dependencies in a .bzl file in the main directory instead of the WORKSPACE, which also makes it easier for others to depend on us (they just need to load deps.bzl and not copy all the individual dependencies).

Reply:
The main issue is that Bazel doesn't support transitive dependencies (which is kind of intentional afaik). This means every project needs to specify all the dependencies, either explicitly or by calling some macro like psi_cardinality_deps. That's also the reason we need preload.bzl, because deps.bzl itself has dependencies.